### PR TITLE
PAM Healthchecks

### PR DESCRIPTION
**Pull request by peteclark-ft**
PAM now checks the health of the push connection (rather than just the `__health` endpoint)
There are also improvements to the healthcheck logic.

### CODE REVIEW

File: feeds/model.go [hunk 1]
```diff
@@ -15,6 +15,7 @@ type Link struct {
 type Feed interface {
 	Start()
 	Stop()
+	IsConnected() bool
```
Comment (khatton-ft) [inline feeds/model.go]: Do we want this as a method in the interface? it's only related to push feeds.

File: feeds/notificationsPushFeed.go [hunk 2]
```diff
@@ -76,9 +87,9 @@ func (f *NotificationsPushFeed) consumeFeed() bool {
 
 		event, err := br.ReadString('\n')
 		if err != nil {
-			infoLogger.Printf("Error: [%v]", err)
-			panic("foo")
-			continue
+			infoLogger.Printf("Error: [%v] - disconnected from push feed!", err)
+			f.connected = false
```
Comment (khatton-ft) [inline feeds/notificationsPushFeed.go]: Strictly speaking, the feed isn't connected, however this method returns. Maybe setting connected to false should be in a func that's deferred at the beginning of the method.

File: config.json.template [hunk 3]
```diff
@@ -45,6 +48,7 @@
         "endpoint": "NOTIFICATIONS_PUSH_URL",
         "granularity": 40,
         "alias": "notifications-push",
+		  "health": "/__notifications-push/__health",
```
Comment (tosan88) [inline config.json.template]: It isn't an issue per se but can you please avoid using tabs? :)

File: healthcheck.go [hunk 4]
```diff
@@ -75,6 +76,32 @@ func (h *Healthcheck) gtg(writer http.ResponseWriter, req *http.Request) {
...
+		Name:             "IsConsumingFromNotificationsPushFeeds",
+		PanicGuide:       pam_run_book_url,
+		Severity:         1,
+		TechnicalSummary: "The connections to the configured notifications-push feeds are operating correctly.",
+		Checker: func() (string, error) {
+			result := true
+			for _, val := range subscribedFeeds {
+				for _, feed := range val {
+					if !feed.IsConnected() {
+						warnLogger.Println("Feed \"" + feed.FeedName() + "\" is not connected!")
```
Comment (khatton-ft) [inline healthcheck.go]: I don't think FeedName is unique when there are multiple delivery environments being monitored.

File: config.json.template (hunk 3 above)
Comment (khatton-ft) [inline config.json.template]: I thought the Go formatting tools enforced tabs! :wow:

Comment (tosan88) [inline config.json.template]: Thank you :)
:+1:

**Pull request merged**